torchprime/torch_xla_models/model/base_causal_lm.py

```diff
@@ -175,10 +116,54 @@ def from_pretrained(self, model_path_or_repo: str):
     if os.path.isdir(model_path_or_repo):
       model_dir = model_path_or_repo
     else:
-      model_dir = snapshot_download(
+      model_dir = huggingface_hub.snapshot_download(
         repo_id=model_path_or_repo, allow_patterns=["*.safetensors*", "config.json"]
       )
 
     # Load weights
-    state_dict = load_safetensors_to_state_dict(model_dir)
+    state_dict = model_utils.load_safetensors_to_state_dict(model_dir)
     self.load_state_dict(state_dict)
+
+  def _maybe_save_checkpoint(self, config: DictConfig) -> None:
```

Comment:
thanks for breaking up a complex method, looks much cleaner now.
No action here, by personal preference is to avoid _maybe, if the logic for _maybe is used in one caller method, I'd rather have it in the caller method.